### ISSUE TITLE
Why the fit loss will change when the model is non trainable with some batch_normalization layers

### ISSUE THREAD

**Issue by geekbutton** [keras-team/keras#9207]
I get a simple CNN model with some batch_normalization layers, and I set the model to non-trainable, then I use fit  to get the loss , but I found that if the batch_size is not equal to 1, then the final loss will be different for each epoch . But if without the batch_normalization layers, the final loss will be the same no matter which the batch_size is.Can anyone explain why will this happen?

**Comment by geekbutton** [comment]
Thank you @punnerud , I know that answer is one of the reason why the loss will change while there are  moving_average_mean and moving_average_std which will change the final loss, but I set the decay = 1.0 in function "assign_moving_average" to make the mean and std keep the same without a moving decay.
 In this condition, if the batch_size = 1 of batch_size = num(samples) then the loss will keep the same, but any other batch_size will make the loss different.
I discuss this with my friends, we think another reason why this will happen is because that the std and mean will be different with different batch_size, and those will influence the final mean loss.
